Additional comments from cmc Wed Mar 3 05:50:05 -0800 2004
Not sure :-), one of the previous changes made along the
cws_srx645_kildarefilterteam22 tag fixed it.
Additional comments from pjanik Wed Mar 3 12:18:08 -0800 2004 
The change that fixed this particular issue is
http://tmp.janik.cz/OpenOffice.org/24978/24978.diff
In case we decide to put this into 1.1.1 as well, there is also complete diff of
sw project between kildarefilterteam22 and its anchor tag.
Index: sw/source/filter/ww8/wrtw8esh.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/wrtw8esh.cxx,v
retrieving revision 1.63.66.1
retrieving revision 1.63.66.1.8.2
diff -u -r1.63.66.1 -r1.63.66.1.8.2
--- sw/source/filter/ww8/wrtw8esh.cxx	9 Jan 2004 14:32:00 -0000	1.63.66.1
+++ sw/source/filter/ww8/wrtw8esh.cxx	3 Feb 2004 09:46:17 -0000	1.63.66.1.8.2
@@ -622,11 +622,17 @@
             }
             else
             {
-                const SwFrmFmt &rPageFmt = 
-                  rWrt.pDoc->GetPageDesc(
-                  rFmt.GetAnchor().GetPageNum()).GetMaster();
-                bBiDi = 
-                  (FRMDIR_HORI_RIGHT_TOP == rPageFmt.GetFrmDir().GetValue());
+                for (USHORT n = rWrt.maFlyPos.Count(); n > 0;)
+                {
+                    const SwPosFlyFrm *pItem = rWrt.maFlyPos[--n];
+                    if (pItem && &pItem->GetFmt() == &rFmt)
+                    {
+                        SwPosition aAnchorPos(pItem->GetNdIndex());
+                        bBiDi = 
+                            (FRMDIR_HORI_RIGHT_TOP == rWrt.pDoc->GetTextDirection(aAnchorPos));
+                        break;
+                    }
+                }
             }
 
             sal_Int32 nLeft = aRect.Left() + aIter->mnThick;
